contrib/devtools/copyright_header.py

```diff
@@ -81,32 +81,32 @@ def compile_copyright_regex(copyright_style, year_style, name):
     return re.compile('%s %s,? %s' % (copyright_style, year_style, name))
 
 EXPECTED_HOLDER_NAMES = [
-    "Satoshi Nakamoto\n",
-    "The Bitcoin Core developers\n",
-    "The Bitcoin Core developers \n",
-    "Bitcoin Core Developers\n",
-    "the Bitcoin Core developers\n",
-    "The Bitcoin developers\n",
-    "The LevelDB Authors\. All rights reserved\.\n",
-    "BitPay Inc\.\n",
-    "BitPay, Inc\.\n",
-    "University of Illinois at Urbana-Champaign\.\n",
-    "MarcoFalke\n",
-    "Pieter Wuille\n",
-    "Pieter Wuille +\*\n",
-    "Pieter Wuille, Gregory Maxwell +\*\n",
-    "Pieter Wuille, Andrew Poelstra +\*\n",
-    "Andrew Poelstra +\*\n",
-    "Wladimir J. van der Laan\n",
-    "Jeff Garzik\n",
-    "Diederik Huys, Pieter Wuille +\*\n",
-    "Thomas Daede, Cory Fields +\*\n",
-    "Jan-Klaas Kollhof\n",
-    "Sam Rushing\n",
-    "ArtForz -- public domain half-a-node\n",
-    "Intel Corporation",
-    "The Zcash developers",
-    "Jeremy Rubin",
+    r"Satoshi Nakamoto\n",
```

Comment:
As I commented on your previous request, turning these into raw strings is not correct. This is **not a pure refactor** and changes the meaning of the code.
```python
>>> r"Satoshi Nakamoto\n" == "Satoshi Nakamoto\n"
False
```

Raw strings are unnecessary here. If you want to avoid the warning stop escaping `.` as `\,`, e.g.
```patch
- "The LevelDB Authors\. All rights reserved\.\n",
+ "The LevelDB Authors\. All rights reserved.\n",
```

Reply:
If you run those strings against `re` instead you'll see something different. E.g. via `re.match`:
```
>>> import re
>>> re.match(".", "K") # standard string, no escape, matches any character
<re.Match object; span=(0, 1), match='K'>
>>> re.match("\.", "K") # standard string, with escape, does not match any character (hence no result)
>>> re.match("\.", ".") # standard string, with escape, does match "."
<re.Match object; span=(0, 1), match='.'>
>>> re.match(r"\.", "K") # raw string, with escape, does not match any character
>>> re.match(r"\.", ".") # raw string, with escape, does match "."
<re.Match object; span=(0, 1), match='.'>
>>> re.match("\\.", "K") # standard string, double escape, does not match any character
>>> re.match("\\.", ".") # standard string, double escape, does match "."
<re.Match object; span=(0, 1), match='.'>
>>> re.match(r"\\.", "K") # raw string, double escape, does not match any character
>>> re.match(r"\\.", ".") # raw string, double escape, does not match "."
```
You can see above that absent the `\`, "." is interpreted as any character, while r"\." and "\\." are both interpreted as an escaped period.

Reply:
Another way to look at this is comparing the script output with master. The output of `contrib/devtools/copyright_header.py report .` from project root currently matches between current master (https://github.com/bitcoin/bitcoin/commit/ab46fe6ec1b37e88c5a06ee7a06ab31cbd18304f) and this branch (https://github.com/bitcoin/bitcoin/pull/15257/commits/8f0b710f4f0e5087eba73cd047883353a25188ac) - but for one character, escaping the `.` in `'Wladimir J\. van der Laan\n'`, in keeping with the above, the matches are the same:
```
-------------------------------------------------------------------------------
667 files examined according to INCLUDE and EXCLUDE fnmatch rules
-------------------------------------------------------------------------------

  18 with zero copyrights
 504 with one copyright
 141 with two copyrights
   4 with three copyrights
   0 with four or more copyrights

-------------------------------------------------------------------------------
Copyrights with dominant style:
e.g. "Copyright (c)" and "<year>" or "<startYear>-<endYear>":

 135 with 'Satoshi Nakamoto\n'
 635 with 'The Bitcoin Core developers\n'
   1 with 'Bitcoin Core Developers\n'
   1 with 'University of Illinois at Urbana-Champaign\.\n'
   6 with 'Pieter Wuille\n'
   3 with 'Wladimir J\. van der Laan\n'
   3 with 'Jeff Garzik\n'
   1 with 'Jan-Klaas Kollhof\n'
   1 with 'Sam Rushing\n'
   2 with 'ArtForz -- public domain half-a-node\n'
   1 with 'Intel Corporation'
   6 with 'The Zcash developers'
   1 with 'Jeremy Rubin'

-------------------------------------------------------------------------------
Copyrights with year list style:
e.g. "Copyright (c)" and "<year1>, <year2>, ...":


-------------------------------------------------------------------------------
Copyrights with no "(c)" style:
e.g. "Copyright" and "<year>" or "<startYear>-<endYear>":

   1 with 'The Bitcoin Core developers\n'
   1 with 'BitPay Inc\.\n'

-------------------------------------------------------------------------------
0 with unexpected copyright holder names
-------------------------------------------------------------------------------
```

Reply:
Okay then I was apparently misunderstanding what this is trying to do in the first place, sorry.

Reply:
All good, thanks!